### PR TITLE
Fixes #25234: Move content uploads to a service class

### PR DESCRIPTION
**Pull request by akofink**
To test (also tests ImportUpload, but this PR doesn't touch that):

```sh
hammer repository upload-content --id 1 --file /tmp/file.txt
```

### CODE REVIEW

File: app/controllers/katello/api/v2/content_uploads_controller.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@ class Api::V2::ContentUploadsController < Api::V2::ApiController
     api :POST, "/repositories/:repository_id/content_uploads", N_("Create an upload request")
     param :repository_id, :number, :required => true, :desc => N_("repository id")
     def create
-      render :json => pulp_content.create_upload_request
+      render :json => Katello::Pulp::Content.create_upload
```
Comment (jlsherrill) [inline app/controllers/katello/api/v2/content_uploads_controller.rb]: I wonder if we should make a Repository#backend_content_service(smart_proxy) method to serve as the layer that determines which service class to use?

Comment (akofink) [reply app/controllers/katello/api/v2/content_uploads_controller.rb]: I thought this was a good idea as well, but then I started thinking about Pulp 3 artifacts and how they aren't associated with a repository. We had talked previously about how Katello will associate artifacts with a repo anyway. I guess we'd just have to make sure we handle the case where the artifact is uploaded the second time for a second repo and already exists as an artifact in Pulp 3.

Comment (jlsherrill) [reply app/controllers/katello/api/v2/content_uploads_controller.rb]: sure, but i wonder if this is a problem today.  What happens if you upload the same rpm to two different repos?

**Comment by jlsherrill** [comment]
[test katello]

File: app/controllers/katello/api/v2/content_uploads_controller.rb (hunk 1 above)
Comment (akofink) [reply app/controllers/katello/api/v2/content_uploads_controller.rb]: I am able to upload the same file to multiple repos and also the same repo multiple times, and pulp gladly accepts it. Seems like Pulp 3 has intentionally improved this behavior over Pulp 2.

**Comment by akofink** [comment]
[test katello]

**Comment by akofink** [comment]
[test katello]

**Comment by akofink** [comment]
Import upload tests are complete. I didn't modify the content uploads controller tests; they function the same after the refactoring. What do you think about content service tests? Useful?

File: app/services/katello/pulp/repository.rb [hunk 2]
```diff
@@ -21,6 +21,25 @@ def self.instance_for_type(repo, smart_proxy)
         Katello::RepositoryTypeManager.repository_types[repo.root.content_type].service_class.new(repo, smart_proxy)
       end
 
+      def import_uploads(uploads, async: nil, generate_metadata: nil, sync_capsule: nil)
+        ForemanTasks.send(async ? :async_task : :sync_task,
+                          ::Actions::Katello::Repository::ImportUpload, @repo, uploads.pluck('id'),
+                          unit_type_id: unit_type_id(uploads), unit_keys: unit_keys(uploads),
+                          generate_metadata: generate_metadata, sync_capsule: sync_capsule)
```
Comment (jlsherrill) [inline app/services/katello/pulp/repository.rb]: The task should be calling into the service class, not the other way around

**Comment by jlsherrill** [comment]
[test katello]

File: app/controllers/katello/api/registry/registry_proxies_controller.rb [hunk 3]
```diff
@@ -374,43 +374,37 @@ def create_tar_file(files, repository, tag)
     def upload_manifest(tar_file)
       upload_id = pulp_content.create_upload_request['upload_id']
       filename = tmp_file(tar_file)
-      File.open(filename, 'rb') do |file|
-        pulp_content.upload_bits(upload_id, 0, file.read)
+      uploads = []
 
-        file.rewind
+      File.open(filename, 'rb') do |file|
```
Comment (beav) [inline app/controllers/katello/api/registry/registry_proxies_controller.rb]: :vhs:

**Comment by beav** [comment]
pr has ack and green tests, merging

thanks @akofink !

**Pull request merged**